RESTResource/RESTResource.js

```diff
@@ -309,20 +342,11 @@ export default class RESTResource {
       }
 
       // params
-      if (typeof options.params === 'object') {
-        options.params = _.mapValues(
-          options.params,
-          param => substitute(param, props, state, this.module, this.logger, this.dataKey)
-        );
-        for (const key of Object.keys(options.params)) {
-          if (options.params[key] === null) {
-            return null;
-          }
-        }
-      } else if (typeof options.params === 'function') {
-        const parsedQuery = queryString.parse(_.get(props, ['location', 'search']));
-        options.params = options.params(parsedQuery, _.get(props, ['match', 'params']), mockProps(state, this.module, props.dataKey, this.logger).resources, this.logger, props);
-      }
+      options.params = buildOption(options.params, props, state, this.module, this.logger, this.dataKey);
+      if (options.params === null) return null;
+
+      // headers
+      options.headers = buildOption(options.headers, props, state, this.module, this.logger, this.dataKey);
```

Comment:
Should we be handling the case where `options.headers` is `null` here similarly to how if `option.params` is?

Reply:
Yeah, I wondered the same. Agreed, @AndrewIsh?

Reply:
Good spot! Since the building of `options.params` and `options.headers` is identical, I've now refactored them. This will also simplify adding further property building in the future should it be required.

Reply:
Actually, scratch that. It seems that returning `null` from `verbActions` in the event that `options.headers` is `null` prevents the request from firing. I'm assuming there's some subtlety there that I'm not seeing. Anyway, I've undone the refactor I detailed above. Sorry for the confusion.